pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -220,6 +159,112 @@ def run_func(self, *args, **kwargs):
 
         return decorator
 
+    def _get_pattern_lst(self, source_path, timestamps=False):
+        '''
+        Get an order list follow the name pattern
+        '''
+        if timestamps:
+            pattern, ext = os.path.basename(
+                os.path.normpath(source_path)).split(".")
+            suffix = "_timestamps.npy"
+        else:
+            pattern, suffix = os.path.basename(
+                os.path.normpath(source_path)).split(".")
+            suffix = "." + suffix
+        path = []
+        for file in os.listdir(os.path.dirname(source_path)):
+            if fnmatch.fnmatch(file, pattern + "*" + suffix):
+                path.append(file)
+        return sorted(path)
+
+    def _get_conatiners_path(self, source_path):
+        return self._get_pattern_lst(source_path)
+
+    def _get_containers(self, index):
+        '''
+        Get container by index, find the next one if the prev one is broken.
+        '''
+        try:
+            container = av.open(self.containers_path[index])
+        except av.AVError:
+            logger.info("The vide is broken")
+            self.current_index += 1
+            return self._get_containers(self.current_index)
+        except IndexError:
+            logger.info("No More Container Found")
+            raise EndofVideoError("Reached end of video file")
+        else:
+            return container
+
+    def _get_timestamps_lst(self, source_path):
```

Comment:
```suggestion
    def _set_timestamps_lst(self, source_path):
```
The function returns `None`